File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        unordered_map<int , int> mapping;
        for(int i =0; i < nums.size(); i++){
            mapping[nums[i]]++;
        }

        vector<vector<int>> freq(nums.size()+1);
        for(auto &pair: mapping){
            freq[pair.second].push_back(pair.first);
        }

        vector<int> result;
        for(int i = freq.size()- 1; i>=0 ; i--){
            for(int num : freq[i]){
                result.push_back(num);
            }
            if(result.size()==k){
                return result;
            }
        } 
        return result;
    }
};

File: Data Structures & Algorithms/majority-element/submission-0.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        chosen_num = None
        count = 0
        for i in nums:
            if count == 0:
                chosen_num = i
            if i != chosen_num:
                count -= 1
            else:
                count  += 1
        return chosen_num